Sources/System/Sockets/SocketAddress.swift

```diff
@@ -99,7 +120,19 @@ extension SocketAddress {
   }
 }
 
+extension Optional where Wrapped == SocketAddress {
+  internal func _withUnsafeBytes<R>(
+    _ body: (UnsafeRawBufferPointer) throws -> R
+  ) rethrows -> R {
+    guard let address = self else {
+      return try body(UnsafeRawBufferPointer(start: nil, count: 0))
```

Comment:
Hmm.... are we `baseAddress == nil` safe? I've pretty much never constructed a `nil` URBP and have certainly assumed that some are not `nil`.

Reply:
We should be! I put guards on count > 0 where it could be a problem. (Well, I hope I handled it everywhere it needs to be handled.)

Reply:
For inter-library pointers, it seems better to me to handle this at the few call sites than to have to always worry about nil base addresses. I haven't wrapped my head around the call-sites, but why doesn't optional chaining work?

Reply:
A nil baseAddress when count == 0 is just a fact of life imo -- the stdlib APIs don't vend optional buffer pointers, so neither should System.